lib/util/result.ts

```diff
@@ -1,5 +1,4 @@
-import type { SafeParseReturnType, ZodType, ZodTypeDef } from 'zod';
-import { ZodError, z } from 'zod';
+import * as z from 'zod';
```

Comment:
I don't like star imports. they block unused code stripping when bundling

Reply:
I don't see any difference for the code we generate:

<img width="1072" alt="image" src="https://github.com/user-attachments/assets/bca3c9e7-810b-4225-8d29-45da5c62ab50" />

<img width="1071" alt="image" src="https://github.com/user-attachments/assets/a609783d-86b9-4ed1-9eca-9bf5b675f381" />

Do we apply any dead code stripping for `node_modules`?

I created this PR to make the code less messy before further refactoring, i.e. to make the zod-specific types be easily detectable in the code.

Reply:
we plan to bundle renovate some time in future and it would be great if dead code would be dropped. I think star imports don't allow it.